Dockershim remove

yanghesong opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
Dockershim has been removed, refer to #97252. Change the corresponding error message of dockershim.
#### Which issue(s) this PR fixes:
Change the corresponding error message of dockershim.
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Change the corresponding error message of dockershim.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

saschagrunert committed:
Remove AppArmor loaded profile validation

In general it could be possible that init containers deploy security
profiles. The existing AppArmor pre-validation would block the complete
workload without this patch being applied. If we now schedule a
workload which contains an unconfined init container, then we will skip
the validation. The underlying container runtime will fail if the
profile is not available after the execution of the init container.

This synchronizes the overall behavior with seccomp.

Signed-off-by: Sascha Grunert <sgrunert@redhat.com>

chymy committed:
Fix link 404

Signed-off-by: chymy <chang.min1@zte.com.cn>

Commit:
fix return code when request /logs with long file name

prateekpandey14 committed:
fix static check of importing the same package multiple times

Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

charlesxsh committed:
fix potential deadlock

pacoxu committed:
kubeadm: detect runtime socket as URL format

- Update defaults for v1beta2 and 3 to have URL scheme
- Raname DefaultUrlScheme to DefaultContainerRuntimeURLScheme
- Prepend a missing URL scheme to user sockets and warn them
that this might not be supported in the future
- Update socket validation to exclude IsAbs() testing
(This is broken on Windows). Assume the path is not empty and has
URL scheme at this point (validation happens after defaulting).
- Use net.Dial to open Unix sockets
- Update all related unit tests

Signed-off-by: pacoxu <paco.xu@daocloud.io>
Signed-off-by: Lubomir I. Ivanov <lubomirivanov@vmware.com>

pacoxu committed:
kube-apiserver in gce: use --api-audiences as --service-account-api-audiences is deprecated

Signed-off-by: pacoxu <paco.xu@daocloud.io>

takotakot committed:
Fix display of CHANGELOG-1.19.md by inserting newline

pacoxu committed:
custom-metrics: do cleanup even if the creation is not full succeed

Signed-off-by: pacoxu <paco.xu@daocloud.io>

oz123 committed:
Build: improve documentation of build artifacts

As described in #100596 this can surprise newcomers.
This adds a paragraph which explains where are the build
artifacts are and how to use the docker images produced
by the build system.

lsytj0413 committed:
fix(*): display the diff as new-old

wzshiming committed:
Request: Add wzshiming as SIG Node reviewer

Commit:
[pkg/printers/internalversion/import_known_versions]: fixup typo

Signed-off-by: Zhou Peng <p@ctriple.cn>

shawnhanx committed:
Add unit tests for registry/apiserverinternal/storageversion

ash2k committed:
Make sleep interruptible

yangjunmyfm192085 committed:
Correct the error 'A lease can only attach up to 10k objects'

Signed-off-by: JunYang <yang.jun22@zte.com.cn>

wking committed:
kubectl: Document --for's Unicode case-folding condition-value comparison

When the wait command was added in 76794643c5 (add wait, 2018-05-18, #64034),
this comparison was via ToLower.  It moved to use Unicode case folding
[1,2] in f4940cf55f (Staticcheck:
vendor/k8s.io/kubectl/pkg/scale|describe/versioned|cmd/top|cmd/util/editor|cmd/top,
2020-01-21, #87403).

[1]: https://pkg.go.dev/strings#EqualFold
[2]: http://www.unicode.org/reports/tr30/tr30-1.html

shawnhanx committed:
Add unit tests for  pkg/registry/authorization/util

justinsb committed:
Use utils string-slice comparison

go-cmp/cmp is intended for tests, and is much less efficient here.

justinsb committed:
Don't perform expensive go-cmp/cmp comparisons if unneeded

The cmp comparison is relatively expensive (#104821).  If we're not
going to log it, we shouldn't make the comparison.

eggiter committed:
fix(cpumanager): Do not release cpus of init containers while they are reused in app containers

andrewsykim committed:
core/v1: stop defaulting Service internalTrafficPolicy when Type=ExternalName

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
core/v1: add validation and defaulting unit tests for when internalTrafficPolicy is nil and Service type is ExternalName

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
core/v1: add unit tests that allow internalTrafficPolicy field when type is ExternalName

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

andrewsykim committed:
core/v1: add comments about why Service internalTrafficPolicy is not forbidden when type is ExternalName

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

Pingan2017 committed:
Remove args enable-swagger-ui since no effect from 1.14

devincd committed:
fix the log description

Signed-off-by: devincd <505259926@qq.com>

Commit:
Update sleep .5 ⇨ sleep 0.5

There is no change in the process.
I changed it because 0.5 seems to be more intuitive than .5.

invidian committed:
k8s.io/apiserver/pkg/server: fix name in APIServerHandler description

In 18177e2bdeafbddeb3d66fec0b8cb88794cd69ff, PostGoRestful field has
been renamed to NonGoRestfulMux, but the documentation change did not
follow.

This commit fixes that to avoid potential confusion.

Signed-off-by: Mateusz Gozdek <mgozdek@microsoft.com>

kerthcet committed:
fix kubelet flushlogs not working together with os.Exit()

Signed-off-by: kerthcet <kerthcet@gmail.com>

gy95 committed:
break the pointless loop when group matched

Signed-off-by: guoyao <1015105054@qq.com>

yxxhero committed:
Optimize Cobra parameters of Controller Manager

Signed-off-by: yxxhero <aiopsclub@163.com>

benluddy committed:
Use a lister for bootstrap flowcontrol config objects.

Instead of listing all flowschemas and prioritylevelconfigurations on
each tick, read from local informer caches.

mysunshine92 committed:
update changelog-1.22

yuanjize committed:
Fix function comment typo

Commit:
Fix and improve comments on kubelet metrics

Commit:
Fix a typo in comment

Signed-off-by: Guangwen Feng <fenggw-fnst@cn.fujitsu.com>

ialidzhikov committed:
Fix the doc string of GetOldObject func

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

ahrtr committed:
replace ioutil with io and os for cluster/images/etcd;
bump the REVISION in Makefile to 5

ahrtr committed:
replace ioutil with io and os for pkg/api

ping035627 committed:
Cleanup FindPluginByName() in VolumePluginMgr

Signed-off-by: PingWang <wang.ping5@zte.com.cn>

update typo

Signed-off-by: PingWang <wang.ping5@zte.com.cn>

update import

Signed-off-by: PingWang <wang.ping5@zte.com.cn>

wojtek-t committed:
P&F: enable mutating work estimator

tklauser committed:
kubectl: remove unnecessary fmt.Sprintf

Directly use the %#v verb in fmt.Fprintf instead.

shawnhanx committed:
fix "k8s.io/apimachinery/pkg/util/validation" is being imported more than once

SergeyKanzhelev committed:
There are no more Feature:Performance tests any longer, removing the fallback option

gy95 committed:
fix duplicate webhook insert operation

Signed-off-by: guoyao <1015105054@qq.com>

SergeyKanzhelev committed:
remove crictl tar

sayaoailun committed:
fix commont of controller manager leadermigration filter

Signed-off-by: sayaoailun <guojianwei007@126.com>

shawnhanx committed:
fix S1017 and S1002 in pkg/apis/core/validation/validation.go

houjun41544 committed:
Fix the wrong log

ash2k committed:
Remove return value from MaybeResetRESTMapper()

erwinvaneyk committed:
Improve the error message when trying to proxy unsupported service type

astraw99 committed:
fix log typo

chendave committed:
Graduate `PreferNominatedNode` to GA

Signed-off-by: Dave Chen <dave.chen@arm.com>

Commit:
Check the private SSH key existence in the node e2e tests

Ahmed-Aghadi committed:
Improve checks for test/e2e/common/storage

ahrtr committed:
add more info when failing to call PdhAddEnglishCounter

Commit:
add unit tests for TestSetNestedStringSlice, TestSetNestedSlice etc.

SergeyKanzhelev committed:
updated deprecation messages from 1.23 to 1.24

wojtek-t committed:
Fix logging resource-scoped watch requests as GET requests.

Commit:
fix obsolete comments in runtime/schema struct

Commit:
Check key existence in the getPrivateSSHKey function

wojtek-t committed:
Update default PF flow schemas to avoid all endpoint/configmaps operations from controller-manager to match leader-election PL

kerthcet committed:
prefer %w to unwrap errors

Signed-off-by: kerthcet <kerthcet@gmail.com>

kebe7jun committed:
Add completion for

victory460 committed:
code cleanup for container/helpers.go

Commit:
kubectl: fix hard-coded value in zsh completion

pmorie committed:
conformance: correct typo for cronjob forbidconcurrent

jonyhy96 committed:
cleanup: remove insecure flag

Signed-off-by: haoyun <yun.hao@daocloud.io>

warmchang committed:
cleanup(apiserver): delete the --max-resource-write-bytes & --json-patch-max-copy-bytes description.

Change-Id: I0fe27dce27db6a1e186bd44a968c6e931de88dd8
Signed-off-by: William Zhang <warmchang@outlook.com>

cyclinder committed:
remove RuntimeClass feature gate
Signed-off-by: cyclinder <qifeng.guo@daocloud.io>

MikeSpreitzer committed:
Note dishonesty of IntervalClock

jonyhy96 committed:
feat: use clock instead

Signed-off-by: haoyun <yun.hao@daocloud.io>

wojtek-t committed:
Slightly improve reliability of storage List tests.

Commit:
Replace hardcoded kubectl with kubectl.Name()

SergeyKanzhelev committed:
support removal of files by the ./hack/update-openapi-spec.sh

Commit:
Fix comment out typo

pacoxu committed:
add probe warning event body like 'Probe terminated redirects'

Commit:
replace syscall with sys/unix pkg in pkg/util/ipvs

gnufied committed:
Add metric for reporting total end-to-end mount time

This metric includes time spent in waiting for devices to be attached,
any RPC calls and performing recursive chown etc.

Commit:
Change the name of the constant

benluddy committed:
Don't bypass ResponseWriter wrappers for apiserver healthz errors.

The effective layering of ResponseWriters is today, from outside to
inside, httplog(timeout(audit(metrics(original)))). From
6e3fd91e1aa3259d7bd67e0a65693e346ade347d, calls to http.Error in the
apiserver's root healthz handler use an unwrapped ResponseWriter --
effectively timeout(audit(metrics(original))) -- to avoid logging
stack traces for those requests.

From 0d50c969c587c8a6c16e0962118305ac652c5a6b, the same call to
http.Error receives a completely-unwrapped ResponseWriter. This has
the effect of bypassing not only the httplog wrapper, but also
timeout, audit, and metrics. The timeout wrapper defends against
the (disallowed) use of underyling ResponseWriter after the completion
of its request's ServeHTTP call. Since that defensive behavior is
being bypassed, it's possible for the root healthz handler to panic
when health probes time out.

Instead of continuing to use a wrapper-aware means of disabling stack
traces, this commit adds a new function to httplog that allows
customization of the stack trace logging predicate on a per-request
basis.

pacoxu committed:
probe: add test case for redirect terminating cases

kerthcet committed:
rename recordSchedulingFailure to handleSchedulingFailure

Signed-off-by: kerthcet <kerthcet@gmail.com>

tkashem committed:
apf: use v1beta2 for registry

yxxhero committed:
add more message for no PodSandbox container

Signed-off-by: yxxhero <aiopsclub@163.com>

hakman committed:
Clean up dockerless build tag

Signed-off-by: Ciprian Hacman <ciprian@hakman.dev>

hakman committed:
Clean up DockerLegacyService interface

Signed-off-by: Ciprian Hacman <ciprian@hakman.dev>

jiekun committed:
#106203 Added test cases for PriorityQueue Activate

danielhelfand committed:
add comment on beta status for APIPriorityAndFairness feature gate

APIPriorityAndFairness reached beta status and is enabled by default in v1.20

Commit:
Remove feature gate  ImmutableEphemeralVolumes

Commit:
Improve test coverage: pkg/api/persistentvolume

jsafrane committed:
Add new watchers to PV controller tests

Add fake Pod and Node watchers to the tests. It only reduces test noise:

Failed to watch *v1.Pod: unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:"", Verb:"watch", Resource:schema.GroupVersionResource{Group:"", Version:"v1", Resource:"pods"}, Subresource:""}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:fields.andTerm{}, ResourceVersion:""}}

jsafrane committed:
Deflake PV metrics unit test

Test 5-7 tries to delete a PVC at the very same time when it detects that
the PV controller started processing the PVC. The controller then sometimes
can't update the PVC and generate an event for it that the test expects.

From PV controller logs (not shown in CI):
> I1221 14:36:34.548160  104481 pv_controller.go:815] updating PersistentVolumeClaim[default/claim5-7] status: set phase Lost failed: cannot update claim claim5-7: claim not found

Typical error in CI:
> FAIL: TestControllerSync (83.22s)
> framework_test.go:202: Event "Warning ClaimLost" not emitted

Therefore wait for the PVC to be fully processed before deleting the PVC to
avoid races.

jsafrane committed:
Fix PV name in unit test

Test 5-5 should use PV with "5-5"i in the name. It makes log analysys
much easier.

derek-pryor committed:
Fix the leak of vSphere client sessions

Implement the test cleanup callback method to close any open vSphere
client connections. Without this clean up sessions were leaking during
testing.

ltagliamonte-dd committed:
add failopen metric

Iceber committed:
if the ok returned by RequestInfoFrom is false, the requestInfo is nil

Iceber committed:
set RemainingItemCount to ParitalObjectMetadataList

hakman committed:
Clean up dockershim in tests

Signed-off-by: Ciprian Hacman <ciprian@hakman.dev>

ialidzhikov committed:
Fix doc for leader-elect-resource-lock flag

Signed-off-by: ialidzhikov <i.alidjikov@gmail.com>

jingxu97 committed:
Change default runtime to containerd for Windows cluster

change default runtime to containerd for windows

Change-Id: I61949ae3687cddb7bfd00715a455ee5eed4da0db

Commit:
Refers to the GC doc in the blockOwnerDeletion API doc

pacoxu committed:
flake fix: remove the error handler for cronjob integration test

Monokaix committed:
Remove unused arg of kubeadm/WaitForKubeletAndFunc

Commit:
fix nil pointer in create secret commands

Commit:
Deprecate service.spec.LoadBalancerIP

Commit:
Update generated files

neolit123 committed:
kubeadm: make "upgrade node" include URL scheme in socket paths

The CRI socket that kubeadm writes as an annotation
on a particular Node object can include an endpoint that
does not have an URL scheme. This is undesired as long term
the kubelet can stop allowing endpoints without URL scheme.

For control plane nodes "kubeadm upgrade apply" takes
the locally defaulted / populated NodeRegistration and refreshes
the CRI socket in PerformPostUpgradeTasks. But for secondary
nodes "kubeadm upgrade node" does not.

Adapt "upgrade node" to fetch the NodeRegistration for this node
and fix the CRI socket missing URL scheme if needed in the Node
annotation.

ltagliamonte-dd committed:
add tests for new metric

yanghesong committed:
CNI parameters should also be deleted with dockershim

yanghesong committed:
Revert "CNI parameters should also be deleted with dockershim"

This reverts commit 5be9c25cc1079f706895e742eff7f6a7a47339d1.

yanghesong committed:
remove dockershim

neolit123 committed:
kubeadm: ensure CRI URL scheme is present in the kubelet env file

During "upgrade node" and "upgrade apply" read the
kubelet env file from /var/lib/kubelet/kubeadm-flags.env
patch the --container-runtime-endpoint flag value to
have the appropriate URL scheme prefix (e.g. unix:// on Linux)
and write the file back to disk.

This is a temporary workaround that should be kept only for 1 release
cycle - i.e. remove this in 1.25.

pacoxu committed:
assignQuota checks if the underlying medium supports quotas and if so, sets

dims commented:
/ok-to-test

dims commented:
/sig node
/assign @adisky @endocrimes

adisky commented on pkg/security/apparmor/validate.go:
```diff
@@ -111,9 +111,9 @@ func validateHost(runtime string) error {
-		return fmt.Errorf("AppArmor is only enabled for 'docker' and 'remote' runtimes. Found: %q", runtime)
+	// Check runtime support. Currently only CRI is supported.
+	if runtime != kubetypes.RemoteContainerRuntime {
+		return fmt.Errorf("AppArmor is only enabled for 'remote' runtimes. Found: %q", runtime)
```
it might sense to remove the whole if conditions, as runtime can only be `remote` in context of kubelet though not sure what values runtime can have in this package

dims replied:
+1 to @adisky 's suggestion.

endocrimes replied:
+1, removing this check entirely seems reasonable to me.

yanghesong replied:
thanks for your suggestion, the whole if conditions has been removed

fasaxc committed:
client-go: Clear the ResourceVersionMatch on paged list calls

API server rejects continuations with ResourceVersionMatch set.

andrewsykim committed:
core/v1: drop Service spec.internalTrafficPolicy on read when type is ExternalName

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

mauriciopoppe committed:
Execute sync before taking the snapshot

andrewsykim committed:
core/v1: update unit tests to not expect internalTrafficPolicy when Service Type=ExternalName

Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

neolit123 committed:
kubeadm: fix panic in runKubeletConfigPhase

The "nro" struct must be initialized before passing it to
GetNodeRegistration().

bostrt committed:
Skip tests unless multizone

MikeSpreitzer committed:
Concurrentize pkg/registry/flowcontrol/ensurer/strategy.go

Stop logging error messages when surprised by concurrent activity.
Concurrent activity is not an error, it is normal.

SataQiu committed:
kubeadm: remove the restriction that the ca.crt can only contain one certificate

pacoxu committed:
Delete .staticcheck_failures

SataQiu committed:
kubeadm: use build tags and split the Windows releated logic into separate files for kubelet component config

riaankleinhans committed:
Update ineligible_endpoints.yaml

ping035627 commented:
/priority backlog
/triage accepted

Commit:
Support IPv6 in echoserver

humblec committed:
csi: correct struct fields in CSI{Persistent}VolumeSource.

The field names in godoc for CSIPersistentVolumeSource and
CSIVolumeSource has been corrected with this commit.

Additional Ref# https://github.com/kubernetes/kubernetes/pull/105963#discussion_r751340196

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

gnufied committed:
rename volume_mount to overall_volume_mount for total metric time

aramperes committed:
Add label selector in 'kubectl rollout' commands

Commit:
cmd/kubectl: make 'kubectl logs' default to the first container when default container cannot be determined or found by annotations (#105964)

* cmd/kubectl: make 'kubectl logs' default to the first container.

While running 'kubectl logs <pod>', If '-c' is omited and the pod has more than one container, and no default container can be determined from annotations, this command shows an error message and exits. With this fix, it defaults to the first container in such scenarios and show its logs. This aligns behavior with what 'kubectl exec' does currently, and is more in line with KEP SIG-CLI 2227 design.

* fix unit test(forgotten)

* fix spelling typo

yanghesong committed:
remove the whole if conditions

Signed-off-by: yanghesong <hesong.yang@foxmail.com>

dims commented:
We need to fix the reference in kubelet.go too right?
https://cs.k8s.io/?q=NewValidator&i=nope&files=.*kubelet.*&excludeFiles=&repos=kubernetes/kubernetes

adisky reviewed (commented):
@yanghesong tests are failing with
```
pkg/kubelet/kubelet.go:834:49: too many arguments in call to apparmor.NewValidator
W0107 03:39:43.384] 	have (string)
W0107 03:39:43.384] 	want ()
```

you need to update `pkg/kubelet/kubelet.go` as well

cyclinder committed:
kube-scheduler: log the unhandled error

Signed-off-by: cyclinder <qifeng.guo@daocloud.io>

yanghesong committed:
Remove runtime in validateHost

Signed-off-by: yanghesong <hesong.yang@foxmail.com>

yanghesong committed:
Remove runtime in validateHost

Signed-off-by: yanghesong <hesong.yang@foxmail.com>

yanghesong committed:
Remove runtime in validateHost

Signed-off-by: yanghesong <hesong.yang@foxmail.com>

tnqn committed:
kube-proxy: fix duplicate port opening

When nodePortAddresses is not specified for kube-proxy, it tried to open
the node port for a NodePort service twice, triggered by IPv4ZeroCIDR
and IPv6ZeroCIDR separately. The first attempt would succeed and the
second one would always generate an error log like below:

"listen tcp4 :30522: bind: address already in use"

This patch fixes it by ensuring nodeAddresses of a proxier only contain
the addresses for its IP family.

yanghesong commented:
> @yanghesong tests are failing with
> 
> ```
> pkg/kubelet/kubelet.go:834:49: too many arguments in call to apparmor.NewValidator
> W0107 03:39:43.384] 	have (string)
> W0107 03:39:43.384] 	want ()
> ```
> 
> you need to update `pkg/kubelet/kubelet.go` as well



> We need to fix the reference in kubelet.go too right? https://cs.k8s.io/?q=NewValidator&i=nope&files=.*kubelet.*&excludeFiles=&repos=kubernetes/kubernetes

Thanks for the heads up, I'll correct it and retest soon.

yanghesong commented:
/retest

yanghesong committed:
Merge remote-tracking branch 'origin/dockershim_remove' into dockershim_remove

# Conflicts:
#	pkg/security/apparmor/validate_test.go

dims commented:
looks like a very bad merge @yanghesong

yanghesong committed:
Merge branch 'master' of github.com:kubernetes/kubernetes into dockershim_remove

 Conflicts:
	pkg/security/apparmor/validate.go
